packages/nimble-components/src/table/specs/table-column-specs/table-column-file-size.md

```diff
@@ -0,0 +1,234 @@
+
+#### Localization of the unit label
+
+This is more difficult, as there does not seem to be a great API for getting a file size unit string for a given locale. The only implemenation I've found is `Intl.NumberFormat`. This function can also do the numeric conversion, but we would not use it for that, because it only supports converting by using a factor of 1000, not 1024. We could instead do the number conversion ourselves and just ask `Intl.NumberFormat` to translate the unit string we need. However, we would have to work around several limitations:
+
+-   Does not have translations for KiB/MiB/etc.
+-   Does not translate plural "bytes".
+-   Throws error for locales with lang subtag `zh_CN` and `zh_TW` -- expects just `zh`.
+-   Requires creating separate `Intl.NumberFormat` instances for each unit we wish to translate.
+-   Missing translations for "byte" in French (gives "o" instead of "octet"), Japanese ("バイト"), Chinese ("字节" in simplified or "位元組" in traditional), Korean ("바이트"), Russian ("Б" instead of "байт"), and probably others.
+
+Options:
+
+1. Find translations of each of our unit labels ("byte", "bytes", "KB", "MB", "GB", "TB", "PB", "KiB", "MiB", "GiB", "TiB", "PiB") for a fixed set of languages we wish to support. Maintain a mapping of language codes (e.g. "fr", "de", "zh_CN") to arrays of those localized unit labels. Given a locale to use, look up its language subtag in our map. If not found, fall back to English.
+
+**Pros:** No work for clients.
+
+**Cons:** Only supports a fixed set of languages. Up to us to find accurate translations. Unsure if subtags other than the language (e.g. region) could be relevant to the translation.
+
+2. Use Nimble label provider. Add label tokens for the units, and rely on clients to provide translations.
+
+**Pros:** Supports any language a client cares to provide translations for.
+
+**Cons:** Less convenient for clients. Does not honor the `lang` setting on the page or on `nimble-theme-provider`.
+
+3. Use `Intl.NumberFormat` and drop support for units based on a factor of 1024.
+
+**Pros:** No translation work for the client or Nimble team. Honors `lang` value.
+
+**Cons:** No support for 1024-based units. Shows "byte" unit even for multiple bytes. Missing several translations for "byte". Unclear if missing translations for other units (KB, MB, etc.).
+
+4. Use `Intl.NumberFormat` and support both 1024 or 1000 based conversions, but just always use KB/MB/GB/TB/PB as the units. There is precedent for this, so it may not be surprising to clients. This is what SLE's current implementation does.
+
+**Pros:** No translation work for the client or Nimble team. Honors `lang` value.
+
+**Cons:** Potentially misleading units label for 1024-based values. Shows "byte" unit even for multiple bytes. Missing several translations for "byte". Unclear if missing translations for other units (KB, MB, etc.).
```

Comment:
nit / fyi add indent spacing so the pros and cons show up under the items in the list

```suggestion
1. Find translations of each of our unit labels ("byte", "bytes", "KB", "MB", "GB", "TB", "PB", "KiB", "MiB", "GiB", "TiB", "PiB") for a fixed set of languages we wish to support. Maintain a mapping of language codes (e.g. "fr", "de", "zh_CN") to arrays of those localized unit labels. Given a locale to use, look up its language subtag in our map. If not found, fall back to English.
   
   **Pros:** No work for clients.
   
   **Cons:** Only supports a fixed set of languages. Up to us to find accurate translations. Unsure if subtags other than the language (e.g. region) could be relevant to the translation.
   
2. Use Nimble label provider. Add label tokens for the units, and rely on clients to provide translations.
   
   **Pros:** Supports any language a client cares to provide translations for.
   
   **Cons:** Less convenient for clients. Does not honor the `lang` setting on the page or on `nimble-theme-provider`.
   
3. Use `Intl.NumberFormat` and drop support for units based on a factor of 1024.
   
   **Pros:** No translation work for the client or Nimble team. Honors `lang` value.
   
   **Cons:** No support for 1024-based units. Shows "byte" unit even for multiple bytes. Missing several translations for "byte". Unclear if missing translations for other units (KB, MB, etc.).
   
4. Use `Intl.NumberFormat` and support both 1024 or 1000 based conversions, but just always use KB/MB/GB/TB/PB as the units. There is precedent for this, so it may not be surprising to clients. This is what SLE's current implementation does.
   
   **Pros:** No translation work for the client or Nimble team. Honors `lang` value.
   
   **Cons:** Potentially misleading units label for 1024-based values. Shows "byte" unit even for multiple bytes. Missing several translations for "byte". Unclear if missing translations for other units (KB, MB, etc.).
```